handler/engine/src/main/java/io/knotx/fragments/engine/Task.java

```diff
@@ -16,20 +16,27 @@
 package io.knotx.fragments.engine;
 
 import io.knotx.fragments.engine.graph.Node;
+import io.vertx.core.json.JsonObject;
 import java.util.Optional;
 
 public class Task {
 
   private final String name;
   private final Node rootNode;
+  private final JsonObject metadata;
```

Comment:
Instead of adding the `metadata` field, please create the `TaskMetadata` class:
```
class TaskMetadata {
NodeMetadata node(String nodeIdentifier) { ... } 
}

class NodeMetadata {
  String getFactory() { ... }
  String getConfig() { ... }
}
```
Then `TaskMetadata` is a part of `FragmentsHandler` but is not passed to `FragmentsEngine`. So the `Task` contract is not modified.

Reply:
Then `TaskProvider` interface can look like:
```
public interface TaskFactory {
  TaskWithMetadata newInstance(FragmentEventContext context);
}
```
where `TaskWithMetadata` may consist of `Task` and `TaskMetadata`.
```
class TaskWithMetadata {
Task task() {...}
TaskMetadata metadata() {...}
} 
```

Reply:
I've considered a similar approach, but there is a problem with matching the data proceesed by `FragmentsEngine` with the original `FragmentEventContext` objects.
I've came up with this solution - `TaskFactory` and `NodeFactory` can optionally enrich a `Task` with metadata, and FragmentsEngine returns the resulting `FragmentEventContexTaskAware` classes (the Task metadata can then be easily accessed). WDYT?

I agree having a dedicated DataObjects for the metadata is desired.
However, the `node(String nodeIdentifier)` accesor seems problematic to me for two reasons:
- introducing `nodeId` is in scope of a different issue
- the graph metadata should reflect the actual tree-like structure of the grapth, for which a JSON representation seems to be the best.